### PR TITLE
sip: add tests for sip over tcp v6

### PR DESCRIPTION
**Pull request by glongo**
## Ticket

If your pull request is related to a Suricata ticket, please provide
the full URL to the ticket here so this pull request can monitor
changes to the ticket status:

Redmine ticket:
https://redmine.openinfosecfoundation.org/issues/3351

### CODE REVIEW

File: tests/sip-tcp-body-frames/test.rules [hunk 1]
```diff
@@ -0,0 +1,11 @@
+alert sip any any -> any any (flow:to_server; frame:pdu; content:"REGISTER"; startswith; sid:2;)
+alert sip any any -> any any (flow:to_client; frame:pdu; content:"SIP/2.0 200 OK|0D 0A|"; startswith; sid:11;)
```
Comment (catenacyber) [inline tests/sip-tcp-body-frames/test.rules:2]: 🤔 maybe we should automate the run of S-V with simulated TCP fragmentation

Comment (glongo) [reply tests/sip-tcp-body-frames/test.rules:2]: TCP data (pkt 4 and 6) in sip-tcp-method/sip-tcp.pcap is already fragmented.

Comment (catenacyber) [reply tests/sip-tcp-body-frames/test.rules:2]: Why does not a PDU frame get created with only packet 4 ?

Comment (catenacyber) [reply tests/sip-tcp-body-frames/test.rules:2]: So, a frame gets created on packet 4 with size 400 (which is wrong)
But no tx gets created.
Then, the right frame gets created on packet 6...

Comment (victorjulien) [reply tests/sip-tcp-body-frames/test.rules:2]: So is this a bug?

Comment (glongo) [reply tests/sip-tcp-body-frames/test.rules:2]: A PDU frame is created only when a packet is parsed correctly, so the behavior described above does not happen anymore.

Comment (catenacyber) [reply tests/sip-tcp-body-frames/test.rules:2]: I think there was a bug in Suricata PR cf https://github.com/OISF/suricata/pull/10037#discussion_r1425882598

Discussion continues now in https://github.com/OISF/suricata/pull/10058#discussion_r1431038422

**Review by victorjulien — APPROVED** [review]

**Comment by victorjulien** [comment]
Merged in #1672, thanks!

**Pull request closed without merging**